### PR TITLE
RPC: gettxout: Slightly improve doc and tests

### PR DESCRIPTION
**Pull request by jtimon**
Slightly related to https://github.com/bitcoin/bitcoin/pull/10822 in the sense that I felt the documentation and testing wasn't as good as it could be while writing it.

Ping @sipa since we discussed this on IRC.

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -944,9 +944,10 @@ UniValue gettxout(const JSONRPCRequest& request)
             "gettxout \"txid\" n ( include_mempool )\n"
             "\nReturns details about an unspent transaction output.\n"
             "\nArguments:\n"
-            "1. \"txid\"       (string, required) The transaction id\n"
-            "2. n              (numeric, required) vout number\n"
-            "3. include_mempool  (boolean, optional) Whether to include the mempool\n"
+            "1. \"txid\"             (string, required) The transaction id\n"
+            "2. \"n\"                (numeric, required) vout number\n"
+            "3. \"include_mempool\"  (boolean, optional) Whether to include the mempool. Default: true."
+            "     Note that an unspent output that is spent in the memepool won't appear.\n"
```
Comment (sipa) [inline src/rpc/blockchain.cpp]: Wow, such meme, many pool.

Comment (jtimon) [inline src/rpc/blockchain.cpp]: hahaha, yeah let's leave a memepool structure for another PR. fixed

**Comment by jtimon** [comment]
btw, hopefully the nit was solved

**Comment by sipa** [comment]
utACK 7ec368927480c809d27de186690f9d6b6c2e9a9f

File: test/functional/wallet.py [hunk 2]
```diff
@@ -56,6 +56,15 @@ def run_test(self):
         assert_equal(len(self.nodes[1].listunspent()), 1)
         assert_equal(len(self.nodes[2].listunspent()), 0)
 
+        self.log.info("test gettxout")
+        confirmed_txid, confirmed_index = utxos[0]["txid"], utxos[0]["vout"]
+        # First, outputs that are unspent both in the chain and in the
+        # mainpool should appear with or without include_mempool
```
Comment (MarcoFalke) [inline test/functional/wallet.py]: mempool

**Comment by MarcoFalke** [comment]
utACK 7ec368927480c809d27de186690f9d6b6c2e9a9f after fixup

**Comment by jtimon** [comment]
Fixed @MarcoFalke 's nit

File: test/functional/wallet.py [hunk 3]
```diff
@@ -56,6 +56,15 @@ def run_test(self):
...
 
+        self.log.info("test gettxout")
+        confirmed_txid, confirmed_index = utxos[0]["txid"], utxos[0]["vout"]
+        # First, outputs that are unspent both in the chain and in the
+        # mempool should appear with or without include_mempool
+        txout = self.nodes[0].gettxout(txid=confirmed_txid, n=confirmed_index, include_mempool=False)
+        assert_equal(txout['value'], 50)
+        txout = self.nodes[0].gettxout(txid=confirmed_txid, n=confirmed_index, include_mempool=True)
+        assert_equal(txout['value'], 50)
+
```
Comment (MarcoFalke) [inline test/functional/wallet.py]: nit: There is trailing whitespace, causing issues with some editors. We don't have this rule in the developer notes, but good to keep in mind for the future.

**Comment by MarcoFalke** [comment]
re-utACK 6d2d2eb

File: src/rpc/blockchain.cpp [hunk 4]
```diff
@@ -944,9 +944,10 @@ UniValue gettxout(const JSONRPCRequest& request)
             "gettxout \"txid\" n ( include_mempool )\n"
             "\nReturns details about an unspent transaction output.\n"
             "\nArguments:\n"
-            "1. \"txid\"       (string, required) The transaction id\n"
-            "2. n              (numeric, required) vout number\n"
-            "3. include_mempool  (boolean, optional) Whether to include the mempool\n"
+            "1. \"txid\"             (string, required) The transaction id\n"
+            "2. \"n\"                (numeric, required) vout number\n"
+            "3. \"include_mempool\"  (boolean, optional) Whether to include the mempool. Default: true."
+            "     Note that an unspent output that is spent in the mempool won't appear.\n"
```
Comment (promag) [inline src/rpc/blockchain.cpp]: Mixed feelings when reading the note, I thought an unspent output is never spent weather the transaction is in the mempool or not, otherwise is not an unspent output.

Suggestion, remove `unspent`: `Note that an output that is spent in the mempool won't appear`.

File: test/functional/wallet.py (hunk 3 above)
Comment (jtimon) [reply test/functional/wallet.py]: Perhaps it is time to add it to the notes. Is is save right now to have your editor configured to remove all trailing whitespaces in any file you save in the project now?
I remember when that wouldn't be save in the sense that your +14-5 commit would turn into +60-50 or something.

File: src/rpc/blockchain.cpp (hunk 4 above)
Comment (promag) [reply src/rpc/blockchain.cpp]: @MarcoFalke it's merged but WDYT?

Comment (MarcoFalke) [reply src/rpc/blockchain.cpp]: "unspent output" refers to the chainstate, whereas "spent in the mempool" has no effect on the chainstate. Imo it is fine.
